src/gui_common/SpeciesDetailsPanelWithFossilisation.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using Godot;
+
+public class SpeciesDetailsPanelWithFossilisation : VBoxContainer
+{
+    [Export]
+    public NodePath SpeciesDetailsPanelPath = null!;
+
+    [Export]
+    public NodePath FossilisationButtonPath = null!;
+
+    [Export]
+    public NodePath FossilisationDialogPath = null!;
+
+#pragma warning disable CA2213
+    private SpeciesDetailsPanel speciesDetailsPanel = null!;
+    private Button? fossilisationButton;
+    private FossilisationDialog fossilisationDialog = null!;
+#pragma warning restore CA2213
+
+    private Species? previewSpecies;
+
+    public Species? PreviewSpecies
+    {
+        get => previewSpecies;
+        set
+        {
+            if (previewSpecies == value)
+                return;
+
+            previewSpecies = value;
+            speciesDetailsPanel.PreviewSpecies = value;
+
+            UpdateFossilisationButtonState();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        speciesDetailsPanel = GetNode<SpeciesDetailsPanel>(SpeciesDetailsPanelPath);
+        fossilisationButton = GetNode<Button>(FossilisationButtonPath);
+        fossilisationDialog = GetNode<FossilisationDialog>(FossilisationDialogPath);
+
+        UpdateFossilisationButtonState();
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            FossilisationButtonPath.Dispose();
```

Comment:
This is incorrect, there must be a null check here to ensure that when Godot partially loads this class, this doesn't cause an exception. If you didn't already please read my recent styleguide changes, those *should* explain this.